apps/accounts/filters.py

```diff
@@ -1,6 +1,95 @@
 from django.contrib.admin import SimpleListFilter
 from django_countries.data import COUNTRIES
+from datetime import datetime
+from django.utils import timezone
 
+class YearDCFilter(SimpleListFilter):
+    title = 'Last updated datacentre'
+    parameter_name = 'last_evidence'
+
+    def last_15_years(self):
+        current_year = datetime.now().year
+
+        return ((f"{current_year - x}", f"{current_year - x}") for x in range(15))
+
+    def lookups(self, request, model_admin):
+        return self.last_15_years()
+
+    def queryset(self, request, queryset):
+        if self.value() is None:
+            return queryset
+
+        year = int(self.value())
+        start_at = datetime(year=year, month=1, day=1)
+        end_at = datetime(year=year+1, month=1, day=1)
+
+        for dt in [start_at, end_at]:
+            if timezone.is_naive(dt):
+                timezone.make_aware(dt)
+
+        return queryset.filter(
+                hostingproviderdatacenter__created_at__gt=start_at
+            ).filter(
+                hostingproviderdatacenter__created_at__lt=end_at
+            )
+
+class YearIPFilter(SimpleListFilter):
+    title = 'Last approved IP Range (sloooowww)'
+    parameter_name = 'last_approved_ip'
+
+    def last_15_years(self):
+        current_year = datetime.now().year
+        return ((f"{current_year - x}", f"{current_year - x}") for x in range(15))
+
+    def lookups(self, request, model_admin):
+        return self.last_15_years()
+
+    def queryset(self, request, queryset):
+        if self.value() is None:
+            return queryset
+
+        year = int(self.value())
+        start_at = datetime(year=year, month=1, day=1)
+        end_at = datetime(year=year+1, month=1, day=1)
+
+        for dt in [start_at, end_at]:
+            if timezone.is_naive(dt):
+                timezone.make_aware(dt)
+
+        return queryset.filter(
+                greencheckipapprove__created__gt=start_at
+            ).filter(
+                greencheckipapprove__created__lt=end_at
+            )
```

Comment:
Generally you can do
```
queryset..filter(
  greencheckipapprove__created__gt=start_at,
  greencheckipapprove__created__lt=end_at
)
```
Which would mean that it is less than date AND greater than other date. 

There is also the `__range` operator https://docs.djangoproject.com/en/dev/ref/models/querysets/#range